### DISCUSSION TITLE
Moose & Mojolicious?

### DISCUSSION THREAD

**Discussion by ajmetz** [mojolicious/mojo#1810] [Q&A] [answered]
Are there any examples of programs using Moose in a Mojolicious Controller, available to look at?
I'm trying it out myself, and so far can only get     
`my  $object     =   MyApp::Controller::MyObject->new(name => 'Bob');`
to work, and not:
`my  $object     =   MyObject->new(name => 'Bob');`
...which gives a "Can't locate object method "new" via package "MyObject" (perhaps you forgot to load "MyObject"?)"

Some context (MyApp is actually MojoTest):
https://pastebin.com/yHxnD19u

Am new to making my own objects. 
I realise Mojo::Base will easily do most of what I want,
except what I want in this instance, is to learn and practice Moose within a Mojolicious Web App, =).

**Comment by Grinnz** [comment, marked as answer]
You always need to use the full package name in order to instantiate objects, this has nothing to do with the object system you choose. You could use something like https://metacpan.org/pod/aliased to make a shorter name available for classes.

**Reply by ajmetz** [reply]
Ah...So everything has been working as it's supposed to this entire time? XD. Lol. Oh my dayz. Thanks for the aliased link - looks good, =D. 

I was used to using Exporter to bring subroutines into a module's namespace via a "use" statement, and then have access to the subroutines with no need to mention the explicit package name, and figured the same would be possible when dealing with objects. Looking at Path::Tiny, I see while that also has Path::Tiny->new is also actually exports 'path', which constructs the object too. Doesn't appear to use Moose though, so not sure if that approach can be taken when using Moose.

**Reply by Grinnz** [reply]
You can export whatever subroutines you want - `path` that you refer to is just a normal exported subroutine which calls the Path::Tiny constructor, Mojo::File exports similar constructors. But this isn't built in behavior for any object system, and it's sometimes confusing to have an object class also export functions.

**Reply by ajmetz** [reply]
Yes, I've just had a chat with a Perl Programming Facebook Group Mentor Scheme Mentor about this. I appreciate your written clarification too, =). 

For now I've also gone back and simplified the boilerplate, as a lot of the extra stuff was there to try and solve the problem, that turned out not to be a problem, lol.
